docker/build.sh

```diff
@@ -56,3 +57,10 @@ if [ $RUN_TESTS = 1 ]; then
     ./tests/bpftrace_test $TEST_ARGS;
   fi
 fi
+
+# Memleak test needs the bpftrace binary with -fsanitize-address
+if [ $RUN_MEMLEAK_TEST = 1 ]; then
+    cmake -DBUILD_ASAN=ON -DBPFTRACE=bpftrace-asan ..
+    make clean && make "$@" -j $(nproc)
+    BPFTRACE_ASAN=$PWD/src/bpftrace-asan ASAN_OPTIONS=alloc_dealloc_mismatch=0 ./tests/memleak-tests.sh
```

Comment:
Should  ASAN_OPTIONS=... instead be inside memleak-tests.sh? That way when someone runs the script manually they get consistent results w/ CI